Sources/UnixSignals/UnixSignal.swift

```diff
@@ -26,6 +26,8 @@ import Dispatch
 /// A struct representing a Unix signal.
 ///
 /// Signals are standardized messages sent to a running program to trigger specific behavior, such as quitting or error handling
+///
+/// - Important: Unix signals are only functional on platforms supporting signals.
```

Comment:
Should we instead compile out these unix signals on Windows for now completely, so even the ServiceGroup doesn't have those parameters in its initializers? We can later add support for the Windows-specific termination handling using the right terminology, and add more initializers then.

Reply:
Once we make this build on Windows, "UnixSignal" becomes API on Windows. I don't love that and would prefer we compile it out and only deliberately add a potential Windows-specific or a cross-platform API (called e.g. "service event") that uses the right platform-specific API under the hood. But landing this is a one-way door we can't remove for the rest of ServiceLifecycle 2.x, and we don't really benefit.

I think it's fine for those writing cross-platform services to handle this in their app for now, and we can introduce a cross-platform event API later.

Reply:
Would `WindowsConsoleEvent` be a symbol that's present on macOS and Linux?

Reply:
Have you considered moving unix signals behind a trait. Trait would be default enabled on all platforms where Swift Service Lifecycle currently builds. Could default to disabled on Windows? Could possibly even annotate them as unavailable?

Reply:
> Yes. Same argument applies the other way around. Somebody wanting to write a cross platform service that uses ServiceGroup should not be forced to write #if os() code.

Fair enough, while I don't agree with the tradeoff here, it is a reasonable position and I'll defer to you here.